stock_landed_costs_delivery/models/purchase_order.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    def _prepare_landed_cost_line_delivery_values(self):
+        tmpl = self.carrier_id.product_id.product_tmpl_id
+        return {
+            "product_id": self.carrier_id.product_id.id,
+            "name": self.carrier_id.name,
+            "account_id": tmpl.get_product_accounts()["stock_input"].id,
+            "price_unit": self.delivery_price,
+            "split_method": "equal",
+        }
+
+    def _prepare_landed_cost_values(self, picking):
+        res = super()._prepare_landed_cost_values(picking)
+        if self.carrier_id.allow_create_landed_cost_lines:
+            if "cost_lines" not in res:
+                res["cost_lines"] = []
```

Comment:
See this abbreviated form for the same:

```suggestion
            res.setdefault("cost_lines", [])
```